pkg/commands/config.go

```diff
@@ -139,6 +154,52 @@ func getBaseImage(platform string, bo *options.BuildOptions) build.GetBase {
 	}
 }
 
+// verifier memoizes digest verification results, so that we avoid redundant
+// digest verification.
+type verifier struct {
+	m       sync.Mutex
+	entries map[string]*entry
+}
+
+// entry is used to hold the sync.Once used to execute the digest verification,
+// and the resulting error (possibly nil), that results from verification.
+type entry struct {
+	once   sync.Once
+	result error
+}
+
+func (v *verifier) verify(ctx context.Context, digest string) error {
+	e := v.getEntry(digest)
+	e.once.Do(func() {
+		// cosign emits a lot of stuff to STDOUT that we don't want
```

Comment:
I know this is just WIP but this is gross, we should fix this upstream before papering over it here.

Reply:
I did, it broke and got rolled back.  I need to sort out what needs to be fixed in my fix :)

Reply:
This is gone from the latest update.  Inlining stuff FTW, I guess.